src/components/overlays/EditPortalRoles/index.tsx

```diff
@@ -112,6 +114,14 @@ export default function EditPortalRoles({ id }: { id: string }) {
       assignedRoles.length === selectedRoles.length &&
       assignedRoles.every((value) => selectedRoles.includes(value)))
 
+  const disabledCheckbox = (currentRole: AppRole) => {
+    const allAdminRoles = allRoles.filter((item) => item.role.includes('Admin'))
```

Comment:
This will be calculated multiple times but always return the same result. Better calculate it once after allRoles is set (L68) using another useState. Then the entire function can be written more compact as:
```
const disabledCheckbox = (currentRole: AppRole) =>
  UserService.getUsername() === id
    ? allAdminRoles.includes(currentRole)
    : false
```

Reply:
@oyo , code updated.
Please check